src/components/sketches/Scene5_happy.ts

```diff
@@ -0,0 +1,290 @@
+      }
+    }
+
+    explode() {
+      for (let i = 0; i < this.explodeSize * EXPLODE_OFFSET; i++) {
+        const particle = new Particle(this.firework.pos.x, this.firework.pos.y, false);
+        this.particles.push(particle);
+      }
+    }
+  }
+
+  class Particle {
+    pos: p5.Vector;
+    firework: boolean;
+    lifespan: number;
+    acc: p5.Vector;
+    ran: number;
+    vel: p5.Vector;
+
+    constructor(x: number, y: number, firework: boolean) {
+      this.pos = p.createVector(x, y);
+      this.firework = firework;
+      this.lifespan = FIREWORK_LIFESPAN;
+      this.acc = p.createVector(0, 0);
+      this.ran = p.random(0, 3);
+      if (this.firework) {
+        this.vel = p.createVector(0, p.random(-12, -8));
+      } else {
+        this.vel = p5.Vector.random2D();
+        this.vel.mult(p.random(2, 10));
+      }
+    }
+
+    applyForce(force: p5.Vector) {
+      this.acc.add(force);
+    }
+
+    update() {
+      if (!this.firework) {
+        this.vel.mult(0.9);
+        this.lifespan -= 4;
+      }
+
+      this.vel.add(this.acc);
+      this.pos.add(this.vel);
+      this.acc.mult(0);
+    }
+
+    done() {
+      if (this.lifespan < 0) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+
+    show() {
+      if (!this.firework) {
+        // after explosion
+        p.strokeWeight(6);
+        p.stroke(255, 211, 97, this.lifespan);
+      } else {
+        p.strokeWeight(8);
+      }
+
+      if (this.ran < 1) p.stroke('rgba(255,211,97,0.2)');
+      else if (this.ran < 2) p.stroke('rgba(255,211,97,0.4)');
+      else p.stroke('rgba(255,211,97,0.6)');
+
+      p.point(this.pos.x, this.pos.y);
+    }
+  }
+
+  class ringParticle {
+    pos: p5.Vector;
+    vel: p5.Vector;
+    acc: p5.Vector;
+    r: number;
+    time: number;
+    explode: boolean;
+    lifespan: number;
+    history: p5.Vector[];
+
+    constructor(x: number, y: number, rot: number) {
+      this.pos = p.createVector(x, y); // Position
+      this.vel = p.createVector(2, 2); // Velocity
+      this.r = RING_RADIUS; //3*noise(this.pos.x/250, this.pos.y/250)*5;
+      this.vel.limit(5);
+      this.acc = p.createVector(0.1, 0.1); // Acceleration
+      this.vel.rotate(rot); // Rotate velocity to the given rotation (avoids complicated maths)
+      this.acc.rotate(rot); // Rotate to the given rotation (avoids complicated maths)
+      this.time = 0; // Time alive
+      this.explode = false;
+
+      this.lifespan = FIREWORK_LIFESPAN;
+      this.history = [];
+    }
+    update() {
+      this.vel.add(this.acc);
+      this.pos.add(this.vel); // Update position with velocity
+      // If older than 40, be deleted next draw event
+      if (this.time <= 40) {
+        // If older than 15, reduce the size of the ringParticles every step
+        if (this.time > 15) {
+          this.r -= 0.05;
+        }
+        this.time++;
+      } else {
+        if (this.lifespan == FIREWORK_LIFESPAN) {
```

Comment:
just in case
```suggestion
        if (this.lifespan <= FIREWORK_LIFESPAN) {
```

Reply:
Oh oi purposely did the section inside the if statement once while th rest of statements in 248 - 257 will run after initial burst when the ring reaches explode radius

Reply:
Hm, is there a way to make sure this doesn't fail if we accidently change the constants s.t. `this.lifespan` will never equal `FIREWORK_LIFESPAN`?

Reply:
yea in line 227 will always init lifespan to FIREWORK_LIFESPAN, so condition keeps it running only once (this.lifespan == FIREWORK_LIFESPAN)